File: extract_links.js
const fs = require('fs');
const path = require('path');

function walk(dir, filelist = []) {
  const files = fs.readdirSync(dir);
  for (const file of files) {
    const filepath = path.join(dir, file);
    if (fs.statSync(filepath).isDirectory()) {
      if (!filepath.includes('node_modules') && !filepath.includes('.git') && !filepath.includes('.next')) {
        filelist = walk(filepath, filelist);
      }
    } else {
      if (filepath.endsWith('.tsx') || filepath.endsWith('.ts')) {
        filelist.push(filepath);
      }
    }
  }
  return filelist;
}

const files = walk('/Users/tonyvelloor/.gemini/antigravity/scratch/weespaces-website/src');
const urlRegex = /href=["'](https?:\/\/[^"']+)["']/g;
const links = new Set();

for (const file of files) {
  const content = fs.readFileSync(file, 'utf8');
  let match;
  while ((match = urlRegex.exec(content)) !== null) {
    links.add(match[1]);
  }
}

console.log(Array.from(links).join('\n'));

File: extract_links.py
import os
import re

directory = '/Users/tonyvelloor/.gemini/antigravity/scratch/weespaces-website/src'
url_pattern = re.compile(r'https?://[^\s\'"<>]+')
links = set()

for root, _, files in os.walk(directory):
    for file in files:
        if file.endswith('.tsx') or file.endswith('.ts'):
            path = os.path.join(root, file)
            with open(path, 'r', encoding='utf-8') as f:
                content = f.read()
                matches = url_pattern.findall(content)
                for match in matches:
                    links.add(match)

for link in sorted(links):
    print(link)
